zulip/zulip
# realm: Add jitsi_server_url feature.

Fixes #17914 

At database level, the default value for jitsi_server_url is empty string, `’’`, as `update_realm()` does not pass parameters’ with value of `None` to `do_set_realm_property` for database update and zulip event systems, unless we add conditional statement like `if k == "jitsi_server_url" and getattr(realm, k) != v:` in the `for loop` (`for k, v in list(req_vars.items()):`) in `update_realm()`.

At the API level, if a client chooses the default value, `update_realm()` will receive `”default”` string and we will parse the value into the empty string `’’`.

`check_string_in_or_url` function is constructed as for validating whether the `jitsi_server_url` parameter passed contained `”default”` or valid URL.

**Testing plan:** 
1. Ensure `realm_jitsi_server_url` is rendered at home view using `class HomeTest`.
2. Added `test_jitsi_server_url` to `class RealmTest` to ensure request calls and responses related to the `jitsi_server_url` parameter are aligned with expectations.
3. Ensure Realm API works properly with `jitsi_server_url` using `class RealmAPITest`
4. Ensure updates on the `jitsi_server_url` property is in sync to other clients using `class RealmPropertyActionTest`
5. Ensure `check_string_in_or_url` is worked as intended.


**GIFs or screenshots:** 
![jitsi_server_url_UI](https://user-images.githubusercontent.com/67208821/117994938-7d50f280-b373-11eb-881d-47e885e4fcdb.gif)
